src/cpu/probes/LoopPoint.py

```diff
@@ -0,0 +1,13 @@
+from m5.params import *
+from m5.objects.Probe import ProbeListenerObject
+
+
+class LoopPoint(ProbeListenerObject):
+
+    type = "LoopPoint"
+    cxx_header = "cpu/probes/looppoint.hh"
+    cxx_class = "gem5::LoopPoint"
+
+    target_pc = VectorParam.UInt64(0, "the target PC")
+    core = Param.BaseCPU(NULL, "the connected cpu")
+    LPmanager = Param.LoopPointManager(NULL, "the looppoint manager")
```

Comment:
```suggestion
    manager = Param.LoopPointManager(NULL, "the looppoint manager")
```

I'd just call this the `manager`. This way you don't have to violate the style with a capital first letter.

Also, what does it mean to be NULL here? Is that valid? If not, then we should just have no default.

Reply:
I got python error when building gem5 if I don't put NULL here. I think NULL is valid when Param is a SimObject.

Reply:
Odd! I would not expect that error. @BobbyRBruce, we should document this behavior or we should figure out if it's a bug. It surprises me that a default value is *required* for SimObject params 

What I meant by valid is whether or not it would be valid for the user to specify NULL. I think not. I believe there would be a fatal or segfault raised. That said, if gem5 requires a default, then that's how it is.

Reply:
It could be I was doing something wrong with other parts. I'll test with only the Param SimObject without default value now.